core/templates/pages/about-page/about-page.component.spec.ts

```diff
@@ -55,165 +60,175 @@ class MockI18nLanguageCodeService {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    loaderService = TestBed.get(LoaderService);
+    userService = TestBed.get(UserService);
   });
-
-  it('should click on credits tab', () => {
-    component.ngOnInit();
-    expect(component.activeTabName).toBe('about');
-
-    component.onTabClick('credits');
-
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#credits');
-    expect(component.activeTabName).toBe('credits');
+  let component;
+  beforeEach(() => {
+    const aboutPageComponent = TestBed.createComponent(AboutPageComponent);
+    component = aboutPageComponent.componentInstance;
   });
 
-  it('should activate about tab on init', () => {
-    windowRef.nativeWindow.location.hash = '#about';
-
-    component.ngOnInit();
+  it('should successfully instantiate the component from beforeEach block',
+    () => {
+      expect(component).toBeDefined();
+      component.ngOnInit();
+    });
 
-    expect(windowRef.nativeWindow.location.hash).toBe('#about');
-    expect(component.activeTabName).toBe('about');
+  it('should get static image url', () => {
+    expect(component.getStaticImageUrl('/path/to/image')).toBe(
+      '/assets/images/path/to/image');
   });
 
-  it('should activate about license on init', () => {
-    windowRef.nativeWindow.location.hash = '#license';
-
+  it('should check user is not logged in and navigate to create lesson page',
+    fakeAsync(() => {
+      const UserInfoObject = {
+        is_moderator: false,
+        is_admin: false,
+        is_super_admin: false,
+        is_topic_manager: false,
+        can_create_collections: false,
+        preferred_site_language_code: null,
+        username: '',
+        email: '',
+        user_is_logged_in: null
+      };
+      spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+        UserInfo.createFromBackendDict(UserInfoObject))
+      );
+      component.ngOnInit();
+      flushMicrotasks();
+      expect(component.userIsLoggedIn).toBe(null);
+      spyOn(
+        siteAnalyticsServiceStub, 'registerCreateLessonButtonEvent')
+        .and.callThrough();
+      spyOn(global, 'setTimeout');
+      component.onClickCreateLessonButton();
+      expect(siteAnalyticsServiceStub.registerCreateLessonButtonEvent)
+        .toHaveBeenCalledWith();
+      expect(component.loginUrl).toBe('/_ah/login');
+    }));
+
+  it('should set component properties when ngOnInit() is called', () => {
     component.ngOnInit();
-
-    expect(windowRef.nativeWindow.location.hash).toBe('#license');
-    expect(component.activeTabName).toBe('foundation');
+    expect(component.userIsLoggedIn).toBe(null);
+    expect(component.classroomUrl).toBe('/learn/math');
+    expect(component.loginUrl).toBe('/_ah/login');
   });
 
-  it('should activate foundation tab on init', () => {
-    windowRef.nativeWindow.location.hash = '#foundation';
-
+  it('should check if loader screen is working', () =>
```

Comment:
`should show and hide loading screen with the correct text`

(Add text to verify the loading screen show and hide)

Reply:
done